neo/rawio/intanrawio.py

```diff
@@ -496,39 +603,54 @@ def read_rhd(filename):
     ordered_channels = []
 
     if version >= V("1.2"):
-        data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = [("timestamp", "int32", BLOCK_SIZE)]
     else:
-        data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = [("timestamp", "uint32", BLOCK_SIZE)]
 
     # 0: RHD2000 amplifier channel
     for chan_info in channels_by_type[0]:
-        name = chan_info["native_channel_name"]
+        name = chan_info["custom_channel_name"]
         chan_info["sampling_rate"] = sr
         chan_info["units"] = "uV"
         chan_info["gain"] = 0.195
         chan_info["offset"] = -32768 * 0.195
         ordered_channels.append(chan_info)
-        data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        else:
+            data_dtype[0] += [(name, "uint16", BLOCK_SIZE)]
```

Comment:
Reading the format again I think the probelm with my data is here. Two points:

1) The first is minor: for the other two formats, [the specification](https://intantech.com/files/Intan_RHD2000_data_file_formats.pdf) mentions that this is int16. Different from the header-attached case where the dtype is uint16. Can you double check that?
2) More importantly. I see no mention in the specification that the data is organized on blocks for the other two formats that we are adding here. For the data that I have, it does not seem to be the case. This is important, because Sam implementation of memmaps assumes that the total file size will be a multiple of the datatypes and if the datatypes includes the block size then this is unlikely to match. If I remove this assumption locally I can succesfully read my files.

That said, I am still curios on why it does work with your files. Is it just a coincidence? I feel that there is something that I am missing. On the other hand, at least for the `one-file-per-channel` case, the structured dtype with block_size is totally unecessary so I don't think there is a point in having it.

I think that following the structure dtype of the header-attached case will pose problems down the line.

Reply:
For 1) You are right, but its not minor. I re-looked at the data and because it is signed it doesn't have an offset so not only does it need to be changed for the dtype it needs to have a different offset/gain setting. THANKS for pointing this out that is huge. 

For 2) You might be right. Let me read more about this tomorrow. I can try to make some more changes and then we can re-test. I'll push some changes for the weekend and then maybe next week we can strategize again.

Reply:
Ok, I think you took care of this. Feel free to mark it as resolved.